src/platform/linux.c

```diff
@@ -567,16 +586,32 @@ void platform_get_user_directory(utf8 *outPath, const utf8 *subDirectory)
 		log_verbose("homedir was null, used getuid, now is = '%s'", homedir);
 		if (homedir == NULL)
 		{
-			log_error("Couldn't find user home directory");
+			log_fatal("Couldn't find user data directory");
+			exit(-1);
 			return;
 		}
 	}
 	char separator[2] = { platform_get_path_separator(), 0 };
 	strncat(buffer, homedir, MAX_PATH);
 	strncat(buffer, separator, MAX_PATH);
+	strncat(buffer, ".config", MAX_PATH);
```

Comment:
`XDG_CONFIG_HOME` should already contain `/.config` part, and in it's absence, `$HOME/.config` should be used as a default value.

This line should stay as is, `/.config` should be added in the `getpwuid` block instead. http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html#variables